src/porepy/numerics/ad/operator_functions.py

```diff
@@ -0,0 +1,463 @@
+
+    def get_jacobian(self, *args: "Ad_array") -> "np.ndarray":
+        result = self.func(*args)
+        return result.jac
+
+
+class LJacFunction(AbstractJacFunction):
+    """
+    Approximates the Jacobian of the function using the L-scheme
+    with a fixed value per dependency.
+    """
+
+    def __init__(
+        self,
+        L: Union[List[float], float],
+        func: Callable,
+        name: str,
+        is_vector_func: Optional[bool] = False,
+    ):
+        """Constructor.
+
+        The L-multiplier for the L-scheme can be passed for every argument specifically
+        using a list.
+        The order in the list has to mach the order of arguments when calling
+        this instance.
+
+        :param L: multiplier for identity for L-scheme
+        :type L: float / List[float]
+        """
+        super().__init__(func, name, is_vector_func)
+        # check and format input for further use
+        if isinstance(L, list):
+            self._L = [float(val) for val in L]
+        else:
+            self._L = [float(L)]
+
+    def get_jacobian(self, *args: "Ad_array") -> "sps.spmatrix":
+        """The approximate jacobian is identity times L.
+
+        Where the respective blocks appears,
+        depends on the total DOFs and the order of arguments passed during the
+        call to this instance.
+        """
+        # the Jacobian of a (Merged) Variable is already a properly sized block identity
+        if len(args) >= 1:
+            jac = args[0].jac * self._L[0]
+
+            # summing identity blocks for each dependency
+            if len(args) > 1:
+                # TODO think about exception handling in case not enough
+                # L-values were provided initially
+                for arg, L in zip(args[1:], self._L[1:]):
+                    jac += arg.jac * L
+        else:
+            # TODO assert zero as scalar will cause no type errors with other operators
+            # this is the case for a function independent of system variables...
+            jac = 0.0
+
+        return jac
+
+
+### FUNCTION DECORATOR ------------------------------------------------------------------------
+
+
+class ADmethod:
+    """
+    Automatic-Differentiation method/function.
+
+    (Decorator) Class for methods representing e.g., physical properties.
+    The decorated function is expected to take scalars/vectors and return a scalar/vector.
+    See example usage below.
+
+    The return value will be an AD operator of a type passed to the decorator.
+
+    EXAMPLE USAGE:
+    .. code-block:: python
+        import porepy as pp
+
+        # decorating class methods
+        class IdealGas:
+
+            @ADmethod(ad_operator=pp.ad.LJacFunction, operators_args={"L"=[1,1]})
+            def density(self, p: float, T: float) -> float:
+                return p/T
+
+        # decorating function
+        @ADmethod(ad_operator=pp.ad.Function)
+        def dummy_rel_perm(s):
+            return s**2
```

Comment:
Your example will throw an error because the argument `name` for `Function` is not provided by this syntax.
And that is the first purpose of this wrapper, parametrization of the used Function class.

Also one has to be careful when using a class as a decorator with or without arguments.
Depending on that, the reference to the function `dummy_rel_perm` will be fed to either `__init__` or `__call__`

And finally, one has two make very important differences between decorating functions and class methods, because of the `self` argument for class methods. Both cases are covered by `ADmethod`.